QMigrate/io/core.py

```diff
@@ -47,7 +73,7 @@ def read_stations(station_file, delimiter=","):
 
     """
 
-    stn_data = pd.read_csv(station_file, delimiter=delimiter)
+    stn_data = pd.read_csv(station_file, **kwargs)
```

Comment:
`*args` and `**kwargs` I believe are syntactic sugar for the Python interpreter saying that the function can take an expandable list of `args` or dict of `kwargs`, rather than having an argument called arg and and argument called kwarg. They are still added to the function scope as a list/dict. So I think you still need to unpack this into the call to `read_csv`. That's my understanding, at least.

Reply:
I've had a play. What matters is that the function gets what it's expecting - if it is defined as:

```myfunc(a_variable, **kwargs)```

then it has to be supplied with `**kwargs` when called. If instead it is defined as:

```myfunc(a_variable, kwargs)```

it expects a single variable `kwargs` which is the `dict` unpacked from the kwargs provided to a parent function.

So in this case `**kwargs` is required, but in the case of `_compute_1d_sweep()` it requires to be supplied with `kwargs`. The mix is what confused me.

(my playing:)
```
In [12]: def myfunc(a, **kwargs): 
    ...:     print(a) 
    ...:     for k, val in kwargs.items(): 
    ...:         print(k, val) 
    ...:                                                                                                   

In [13]: myfunc(a=1, b=2, c=3)                                                                             
1
b 2
c 3

In [14]: def another_func(a, **kwargs): 
    ...:     print(a) 
    ...:     for k, val in kwargs.items(): 
    ...:         print(k, val) 
    ...:                                                                                                   

In [15]: def myfunc(a, **kwargs): 
    ...:     print(a) 
    ...:     for k, val in kwargs.items(): 
    ...:         print(k, val) 
    ...:     another_func(a, kwargs) 
    ...:                                                                                                   

In [16]: myfunc(a=1, b=2, c=3)                                                                             
1
b 2
c 3
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-16-e8f106c5d38c> in <module>
----> 1 myfunc(a=1, b=2, c=3)

<ipython-input-15-46917818eca1> in myfunc(a, **kwargs)
      3     for k, val in kwargs.items():
      4         print(k, val)
----> 5     another_func(a, kwargs)
      6 

TypeError: another_func() takes 1 positional argument but 2 were given

In [17]: def myfunc(a, **kwargs): 
    ...:     print(a) 
    ...:     for k, val in kwargs.items(): 
    ...:         print(k, val) 
    ...:     another_func(a, **kwargs) 
    ...:                                                                                                   

In [18]: myfunc(a=1, b=2, c=3)                                                                             
1
b 2
c 3
1
b 2
c 3

In [19]: def another_func(a, kwargs): 
    ...:     print(a) 
    ...:     for k, val in kwargs.items(): 
    ...:         print(k, val) 
    ...:                                                                                                   

In [20]: myfunc(a=1, b=2, c=3)                                                                             
1
b 2
c 3
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-20-e8f106c5d38c> in <module>
----> 1 myfunc(a=1, b=2, c=3)

<ipython-input-17-669bb3557823> in myfunc(a, **kwargs)
      3     for k, val in kwargs.items():
      4         print(k, val)
----> 5     another_func(a, **kwargs)
      6 

TypeError: another_func() got an unexpected keyword argument 'b'

In [21]: def myfunc(a, **kwargs): 
    ...:     print(a) 
    ...:     for k, val in kwargs.items(): 
    ...:         print(k, val) 
    ...:     another_func(a, kwargs) 
    ...:                                                                                                   

In [22]: myfunc(a=1, b=2, c=3)                                                                             
1
b 2
c 3
1
b 2
c 3
```

Reply:
Yeah, think that's my bad for not being consistent - should be `**kwargs` in `_compute_1d_sweep` and unpack the kwargs in the call to that function.